backend/src/main/java/pl/edu/pw/elka/etherscan/RealEtherscanConnector.java

```diff
@@ -3,17 +3,29 @@
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.client.RestTemplate;
 import pl.edu.pw.elka.etherscan.dtos.EtherscanTransactionsDto;
+import pl.edu.pw.elka.minedBlocks.dtos.MinedBlocksDto;
 
 class RealEtherscanConnector implements EtherscanConnector {
 
     private final static String BASE_API_URL =
             "http://api.etherscan.io/api?module=account&action=txlist&startblock=0&endblock=99999999&sort=asc&address=";
 
+    private final static String MINED_BLOCKS_URL =
+            "https://api.etherscan.io/api?module=account&action=getminedblocks&blocktype=blocks&address=";
+
     @Override
     public EtherscanTransactionsDto getTransactions(String address) {
         RestTemplate restTemplate = new RestTemplate();
         String apiUrl = BASE_API_URL + address;
         ResponseEntity<EtherscanTransactionsDto> entity = restTemplate.getForEntity(apiUrl, EtherscanTransactionsDto.class);
         return entity.getBody();
     }
+
+    @Override
+    public MinedBlocksDto getMinedBlocks(String address) {
+        RestTemplate restTemplate = new RestTemplate();
```

Comment:
warto to wydzielic do konstruktora i zarowno tutaj jak i wyzej nie tworzyc za kazdym razem nowej instancji